### ISSUE TITLE
Test suite hangs with gnupg 2.0

### ISSUE THREAD

**Issue by remicollet** [php-gnupg/php-gnupg#3]
On RHEL / CentOS we only have gnupg v2 (no gnupg v1)

**With version 1.3.6** the test suite have lot of failure (everywhere the passphrase is needed, I'm working on this issue).

**With version 1.4.0RC1** the test suite simply hangs (in tests/gnupg_oo_encrypt.phpt)

Fails on Fedora 22 (gpgme 1.4.3, gnupg 2.1.11)
Hangs on Fedora 21 (gpgme 1.4.3, gnupg 2.0.29)
Hangs on RHEL 7 (gpgme 1.3.2, gnupg 2.0.22)

So I really think this is related to gnupg version 2.0 vs 2.1

Failure seems acceptable, when hang doesn't.

**Comment by remicollet** [comment]
I forget to say, I haven't found any explanation in the diff between versions 1.3.6 and 1.4.0RC1 :(

**Comment by bukka** [comment]
I'm actually using fedora. Have you tried to compile with:

```
$ ./configure --with-gpg=/usr/bin/gpg
```

By default it will use gpg2 which is not supported atm.

**Comment by bukka** [comment]
anyway I will take a look why it hangs as it should be exactly the same as 1.3.6. The only point of this release is to support PHP 7 and possibly fix some compilation issue from the previous version. All improvements and test fixes (more rewrite probably as they are dependent and not really good) will go to 1.5

**Comment by remicollet** [comment]
> I'm actually using fedora. Have you tried to compile with
Yes, as I said, no problem with gnupg 1 ;)

The issue in only on RHEL which only have gpg2 (/usr/bin/gpg is a symlink to gpg2), but can be reproduced on Fedora <= 22

**Comment by remicollet** [comment]
FYI, here is the minimal C app I use to reproduce the hang issue (could be simpler)
https://github.com/remicollet/remirepo/blob/master/php/pecl/php-pecl-gnupg/info.c

To reproduce the hang, with gnupg 2.0, and without tty of X display

      DISPLAY= ./info /usr/bin/gpg2  2>&1 | cat

**Comment by bukka** [comment]
Nice. That's very helpful. Thanks. Will hopefully find some time to look into it later

**Comment by remicollet** [comment]
Sorry about the noise... this is not a regression.

After digging for hour... 1.3.6 also hangs in some case.

php run-tests.php ... => fails
php -n run-tests.php... => hangs

Perhaps a simple note in the README about "gnupg 2 not supported" is enough.
Feel free to close this bug.

FYI, php-pecl-gnupg is now available in my repository, and use gnupg1 (I have finally choose to build a RPM of gnupg 1.4.20 which can be installed beside system gnupg 2.x)

**Comment by bukka** [comment]
Thanks. I have updated README and set milestone for this to 1.5 to make sure it won't hang there.

**Comment by travisghansen** [comment]
This same test (meaning encrypting and then decrypting with a key that has a passphrase) is part of my test suite here: https://github.com/travisghansen/phpgpg

It works for me on both Arch with
```
gnupg 2.1.12-2
gpgme 1.6.0-2
php-gnupg 1.4.0RC1-1
```
And centos7 php and pals installed form @remicollet 's repo.

Not sure if I've got something messed up on my test but it appears clean.

**Comment by bukka** [comment]
Fixed since 1.5.0RC1
